R/plot_section.R

```diff
@@ -0,0 +1,153 @@
+#' Generate stratigraphic section plot
+#'
+#' A function to plot the stratigraphic ranges of fossil taxa from occurrence
+#' data.
+#'
+#' @param occdf \code{dataframe}. A dataframe of fossil occurrences containing
+#' at least two columns: names of taxa, and their stratigraphic position
+#' (see `name` and `level` arguments).
+#' @param name \code{character}. The name of the column you wish to be treated
+#' as the input names, e.g. "genus" (default).
+#' @param level \code{character}. The name of the column you wish to be treated
+#' as the stratigraphic levels associated with each occurrence, e.g. "bed"
+#' (default) or "height".
+#' @param certainty \code{character}. The name of the column you wish to be
+#' treated as the information on whether an identification is certain ("1") or
+#' uncertain ("0"). By default, no column name is provided, and all occurrences
+#' are assumed to be certain. In the plot, any occurrence labelled as certain
+#' will be plotted with a black circle, while any occurrence labelled as
+#' uncertain will be plotted with a white circle.
+#' @param by \code{character}. How should the output be sorted?
+#' Either: "FAD" (first appearance; default), "LAD" (last appearance), or
+#' "name" (alphabetically by taxon names).
+#' @param units \code{character}. Should the y-axis be labelled as "height"
+#' (stratigraphic height in metres) or "beds" (beds numbered from bottom to
+#' top)?
+#' @param label \code{character}. The title given to the plot, e.g. "Section A"
+#' (default).
+#'
+#' @return No return value. Function is used for its side effect, which is to
+#' create a plot showing the stratigraphic ranges of taxa in a section,
+#' with levels at which the taxon was sampled indicated with a point.
+#'
+#' @section Developer(s):
+#' Bethany Allen & Alexander Dunhill
+#' @section Reviewer(s):
+#' William Gearty & Lewis A. Jones
+#'
+#' @examples
+#' # Sample tetrapod occurrences
+#' tetrapod_names <- tetrapods$accepted_name[1:50]
+#' # Simulate bed numbers
+#' beds_sampled <- sample.int(n = 10, size = 50, replace = TRUE)
+#' # Simulate certainty values
+#' certainty_sampled <- sample(x = 0:1, size = 50, replace = TRUE)
+#' #Combine into data frame
+#' occdf <- data.frame(taxon = tetrapod_names, bed = beds_sampled,
+#' certainty = certainty_sampled)
+#' # Plot stratigraphic ranges
+#' par(mar = c(12, 5, 2, 2))
+#' tax_range_strat(occdf)
```

Comment:
I think this needs updating? The function is currently called `plot_section`, though I think I prefer `tax_range_strat`.